pinot-core/src/main/java/org/apache/pinot/core/common/datablock/DataBlockUtils.java

```diff
@@ -55,8 +47,17 @@ public static MetadataBlock getErrorDataBlock(Exception e) {
     return errorBlock;
   }
 
-  public static MetadataBlock getEmptyDataBlock(DataSchema dataSchema) {
-    return dataSchema == null ? EOS_DATA_BLOCK : new MetadataBlock(dataSchema);
+  public static BaseDataBlock getErrorDataBlock(Map<Integer, String> exceptions) {
+    MetadataBlock errorBlock = new MetadataBlock(EMPTY_SCHEMA);
+    for (Map.Entry<Integer, String> exception : exceptions.entrySet()) {
+      errorBlock.addException(exception.getKey(), exception.getValue());
+    }
+    return errorBlock;
+  }
+
+  public static MetadataBlock getEmptyDataBlock(@Nonnull DataSchema dataSchema) {
```

Comment:
What is the difference between this and EOS block ?

Reply:
this was a naming error. it should be end-of-stream metadata block. let me fix it